hadoop-cloud/src/hadoop-3/main/scala/org/apache/spark/internal/io/cloud/PathOutputCommitProtocol.scala

```diff
@@ -0,0 +1,263 @@
+   * to commit, even when a store has listing inconsistency (i.e. Amazon S3),
+   * there is no risk of that affecting the operation.
+   * Similarly, because the superclass's implementation of this method uses
+   * a UUID in the source name, eventual consistency on an update will not be
+   * an issue.
+   * There is a very small risk that Amazon S3's brief caching of 404s
+   * can cause any probes for the file to fail if the task is committed
+   * immediately after the file was created.
+   *
+   * @param taskContext task
+   * @param absoluteDir destination dir
+   * @param ext         extension
+   * @return an absolute path
+   */
+  override def newTaskTempFileAbsPath(
+      taskContext: TaskAttemptContext,
+      absoluteDir: String,
+      ext: String): String = {
+
+    val file = super.newTaskTempFileAbsPath(taskContext, absoluteDir, ext)
+    logInfo(s"Creating temporary file $file for absolute path for dir $absoluteDir")
+    file
+  }
+
+  /**
+   * Build a filename which is unique across all task events.
+   * It does not have to be consistent across multiple attempts of the same
+   * task or job.
+   *
+   * @param taskContext task context
+   * @param ext         extension
+   * @return a name for a file which must be unique across all task attempts
+   */
+  protected def buildFilename(
+      taskContext: TaskAttemptContext,
+      ext: String): String = {
+
+    // The file name looks like part-00000-2dd664f9-d2c4-4ffe-878f-c6c70c1fb0cb_00003-c000.parquet
+    // Note that %05d does not truncate the split number, so if we have more than 100000 tasks,
+    // the file name is fine and won't overflow.
+    val split = taskContext.getTaskAttemptID.getTaskID.getId
+    f"part-$split%05d-$jobId$ext"
+  }
+
+  override def setupJob(jobContext: JobContext): Unit = {
+    logDebug("setup job")
+    super.setupJob(jobContext)
+  }
+
+  override def commitJob(
+      jobContext: JobContext,
+      taskCommits: Seq[FileCommitProtocol.TaskCommitMessage]): Unit = {
+    logDebug(s"commit job with ${taskCommits.length} task commit message(s)")
+    super.commitJob(jobContext, taskCommits)
+  }
+
+  /**
+   * Abort the job; log and ignore any IO exception thrown.
+   * This is invariably invoked in an exception handler; raising
+   * an exception here will lose the root cause of the failure.
+   *
+   * @param jobContext job context
+   */
+  override def abortJob(jobContext: JobContext): Unit = {
+    try {
+      super.abortJob(jobContext)
+    } catch {
+      case e: IOException =>
+        logWarning("Abort job failed", e)
+    }
+  }
+
+  /**
+   * Abort the task; log and ignore any failure thrown.
+   * This is invariably invoked in an exception handler; raising
+   * an exception here will lose the root cause of the failure.
+   * @param taskContext context
+   */
+  override def abortTask(taskContext: TaskAttemptContext): Unit = {
+    logDebug("Abort task")
```

Comment:
See previous comments about unhelpful debug messages for task-level events.

(And you always have `logTrace`, too...)

Reply:
How about: logTrace with job/taskID? I know that spark JobIds are not as unique as MR ones, but it will give more context to the messages in a worker doing >1 job? Same for error messages generated when catching IOEs in job/task abort...task attempt IDs are potentially useful here

Reply:
> logTrace with job/taskID

That sounds ok. Task ID is probably enough (if you mean `TaskContext.get.taskAttemptId`).

Reply:
I've cut all this completely. the s3a committers are fairly detailed if you ask them to log, and if you really need it, you can tell the AWS SDK to print the HTTP communications